src/Components/Components/src/Routing/NavigationContext.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Routing
+{
+    public class NavigationContext
```

Comment:
Should this be sealed?

Reply:
I didn't anticipate that it would be useful for people to subclass this but we can make this explicit with the modifier. Good call!